tests/tensor_search/test_score_modifiers_search.py

```
@@ -0,0 +1,551 @@
+                                    non_tensor_fields=["multiply_1", "multiply_2", "add_1", "add_2",
+                                                       "filter"], auto_refresh=True)
+
+        normal_res = tensor_search.search(config=self.config, index_name=self.index_name,
+                                          text="what is the rider doing?", score_modifiers=None, result_count=10)
+        normal_score = normal_res["hits"][0]["_score"]
+
+        epsilon = 1e-5
+        score_modifiers_list = [
+            {
+                # miss one weight
+                "multiply_score_by":
+                    [{"field_name": "multiply_1",
+                      "weight": 1,},
+                     {"field_name": "multiply_2",}],
+                "add_to_score": [
+                    {"field_name": "add_1", "weight" : -3,
+                     },
+                    {"field_name": "add_2", "weight": 1,
+                     }]
+            },
+            {
+                # 0 weight
+                "multiply_score_by":
+                    [{"field_name": "multiply_1",
+                      "weight": 1.0, },
+                     {"field_name": "multiply_2",
+                      "weight": 0}],
+                "add_to_score": [
+                    {"field_name": "add_1", "weight": -3,
+                     },
+                    {"field_name": "add_2", "weight": 1,
+                     }]
+            },
+            {
+                # remove one field
+                "multiply_score_by":
+                    [
+                     {"field_name": "multiply_2",
+                      "weight": 1.2}],
+                "add_to_score": [
+                    {"field_name": "add_1", "weight": -3,
+                     },
+                    {"field_name": "add_2", "weight": 1,
+                     }]
+            },
+            {  # one part to be none
+                "multiply_score_by": [],
```

Would validation prevent this from working?

---

updated. it should raise an error now.